src/sql/connection/connection.py

```diff
@@ -615,124 +456,266 @@ def get_curr_identifiers(self) -> list:
+            if detailed_msg is not None:
+                raise exceptions.UsageError(detailed_msg)
+            else:
+                print(e)
+        except Exception as e:
+            raise cls._error_invalid_connection_info(e, connect_str) from e
+
+    @classmethod
+    def _error_invalid_connection_info(cls, e, connect_str):
+        err = UsageError(
+            "An error happened while creating the connection: "
+            f"{e}.{_suggest_fix(env_var=False, connect_str=connect_str)}"
+        )
+        err.modify_exception = True
+        return err
 
 
-class DBAPIConnection(Connection):
-    """
-    A connection object for generic DBAPI connections
-    """
+class DBAPIConnection(AbstractConnection):
+    """A connection object for generic DBAPI connections"""
+
+    is_dbapi_connection = True
 
     @telemetry.log_call("DBAPIConnection", payload=True)
-    def __init__(self, payload, engine=None, alias=None):
+    def __init__(self, payload, connection, alias=None):
         try:
-            payload["engine"] = type(engine)
+            payload["engine"] = type(connection)
         except Exception as e:
             payload["engine_parsing_error"] = str(e)
 
-        if engine is None:
-            raise ValueError("Engine cannot be None")
-
         # detect if the engine is a native duckdb connection
-        _is_duckdb_native = _check_if_duckdb_dbapi_connection(engine)
-        connection_name_ = "duckdb" if _is_duckdb_native else "custom_driver"
+        _is_duckdb_native = _check_if_duckdb_dbapi_connection(connection)
 
-        self.url = str(engine)
-        self.name = connection_name_
-        self.dialect = connection_name_
-        self.session = DBAPISession(self, engine)
+        self._dialect = "duckdb" if _is_duckdb_native else None
 
-        self.connections[alias or connection_name_] = self
+        self._connection = connection
+        self._connection_class_name = type(connection).__name__
 
-        self.connect_args = None
-        self.alias = alias
-        Connection.current = self
+        # calling init from AbstractConnection must be the last thing we do as it
+        # register the connection
+        super().__init__(alias=alias or self._connection_class_name)
 
-        # TODO: create an abstract class
-        self._result_sets = []
+        # TODO: delete this
+        self.name = self._connection_class_name
+
+    @property
+    def dialect(self):
+        return self._dialect
+
+    def raw_execute(self, query):
+        cur = self.connection.cursor()
+        cur.execute(query)
+        return cur
+
+    def _get_database_information(self):
+        return {
+            "dialect": self.dialect,
+            "driver": self._connection_class_name,
+            "server_version_info": None,
+        }
+
+    @property
+    def url(self):
+        """Returns None since DBAPI connections don't have a url"""
+        return None
+
+    @property
+    def connection_sqlalchemy(self):
```

Comment:
no, we need this because both SQLAlchemyConnection and DBAPIConnection have a connection attribute but only the format has a connection_sqlalchemy.

So if any feature depends on sqlalchemy, you can do: `connection.connection_sqlalchemy` and it'll work if the user has a sqlalchemyconenction but it'll throw an error "this feature is only supported for sqlalchemy connections" if using a DBAPIConnection, this prevents the need to add conditionals everywhere to check whether you have a sqlalchemy connection or not